### DISCUSSION TITLE
How to add spans to @service classes in Java spring boot?

### DISCUSSION THREAD

**Discussion by sajjanbalar** [getsentry/sentry-java#2197] [General]
I want to enable sentry tracing via spans for my backend server which is written in Java Spring.
There is a class called ESService with an @Service annotation. The @SentrySpan annotation has been added to this class but I am not seeing the perf times of the methods in this class in Sentry. 

What do I need to do?

**Comment by bruno-garcia** [comment]
Spans are added to Transactions. Do you have a transaction happening? If that service is the entry point (like it's invoked by the framework or something) you could annotate with `@SentryTransaction` instead. Otherwise maybe you haven't setup performance yet? https://docs.sentry.io/platforms/java/guides/spring-boot/performance/

For incoming Web requests, the SDK automatically starts a transaction. So any `@SentrySpan` that happens in a request should automatically get appended to the it
